version: '4.2'
services:
  haproxy:
    image: haproxy:2.6
    volumes:
      - ./docker/haproxy.cfg:/usr/local/etc/haproxy/haproxy.cfg
      - ./certs:/usr/share/auth2/certs
    ports:
      - 80:80
      - 443:443
    sysctls:
      - net.ipv4.ip_unprivileged_port_start=0
  auth2:
    image: paasim/auth2
    build: docker/
    volumes:
      - ./target/debug/auth2:/usr/bin/auth2
      - ./static:/usr/share/auth2/static
      - ./certs:/usr/share/auth2/certs
      - ./certs.db:/var/lib/auth2/certs.db
    ports:
      - 3377:3377
    env_file: ./deb/env
    init: true
  python:
    image: python:3-bookworm
    volumes:
      - ./docker/server.py:/server.py
    command: /server.py
    ports:
      - 8080:8080
    init: true
